src/components/application_manager/test/commands/hmi/sdl_get_user_friendly_message_request_test.cc

```diff
@@ -0,0 +1,173 @@
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdint.h>
+#include <string>
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "application_manager/commands/hmi/sdl_get_user_friendly_message_request.h"
+#include "application_manager/mock_application.h"
+#include "application_manager/mock_state_controller.h"
+#include "application_manager/policies/mock_policy_handler_interface.h"
+#include "application_manager/mock_message_helper.h"
+#include "application_manager/mock_hmi_capabilities.h"
+#include "smart_objects/smart_object.h"
+#include "commands/command_request_test.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace hmi_commands_test {
+
+using application_manager::commands::MessageSharedPtr;
+using application_manager::commands::SDLGetUserFriendlyMessageRequest;
+using application_manager::MockMessageHelper;
+using test::components::policy_test::MockPolicyHandlerInterface;
+using test::components::application_manager_test::MockHMICapabilities;
+using testing::_;
+using testing::Return;
+using testing::ReturnRef;
+
+namespace {
+const uint32_t kCorrelationID = 1u;
+const uint32_t kAppID = 2u;
+const std::string kLanguageDe = "de-de";
+const std::string kLanguageEn = "en-gb";
+const std::string kMessageCodes = "messageCodes";
+}  // namespace
+
+namespace strings = ::application_manager::strings;
+
+class SDLGetUserFriendlyMessageRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  SDLGetUserFriendlyMessageRequestTest()
+      : mock_message_helper_(*MockMessageHelper::message_helper_mock()) {}
+
+ protected:
+  void SetUp() OVERRIDE {
+    mock_app_ = CreateMockApp();
+  }
+
+  void InitCommand(const uint32_t& timeout) OVERRIDE {
+    CommandRequestTest<CommandsTestMocks::kIsNice>::InitCommand(timeout);
+    ON_CALL((*mock_app_), app_id()).WillByDefault(Return(kAppID));
+    EXPECT_CALL(app_mngr_, application_by_hmi_app(kAppID))
+        .WillOnce(Return(mock_app_));
+  }
+  MockAppPtr mock_app_;
+  MockPolicyHandlerInterface mock_policy_handler_;
+  MockMessageHelper& mock_message_helper_;
+};
+
+TEST_F(SDLGetUserFriendlyMessageRequestTest, Run_LanguageSet_SUCCESS) {
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::correlation_id] = kCorrelationID;
+  (*msg)[strings::msg_params][strings::app_id] = kAppID;
+
+  (*msg)[strings::msg_params][kMessageCodes] =
+      SmartObject(smart_objects::SmartType_Array);
+  (*msg)[strings::msg_params][kMessageCodes][0] = SmartObject(kLanguageDe);
+  (*msg)[strings::msg_params][kMessageCodes][1] = SmartObject(kLanguageEn);
+
+  (*msg)[strings::msg_params][strings::language] =
+      hmi_apis::Common_Language::EN_GB;
```

Comment:
@pvvasilev First `hmi_apis::Common_Language::EN_GB` is assigned to `(*msg)[strings::msg_params][strings::language]` then it is used in `CommonLanguageToString(hmi_apis::Common_Language::EN_GB)` for a check. On the both places it should be the same. However in this implementation this is not guaranteed because if the assignment is changed, the check is not updated - and vice versa. So it is better to create a `const hmi_apis::Common_Language::eType` and use in both in assignment and check.